src/codec.rs

```diff
@@ -20,8 +20,7 @@ impl From<Error> for CodecError {
     // TODO: tbd in #83, also this direction shouldn't be necessary.
     fn from(e: Error) -> CodecError {
         match e {
-            Error::DecodingError => CodecError::DecodingError,
-            Error::UnsupportedCiphersuite => CodecError::DecodingError,
+            _ => CodecError::DecodingError,
```

Comment:
Couldn't this also be an `EncodingError`?

Reply:
This direction really shouldn't exist.. I guess I'll start on #83 at some point ... this getting annoying.